tests/lib/json fails on cc3220sf_launchxl

```
In file included from /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/json/json.c:8:0:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/json/json.c: In function ‘lexer_string’:
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:58:54: error: array subscript has type ‘char’ [-Werror=char-subscripts]
 #define __ctype_lookup(__c) ((__ctype_ptr__+sizeof(""[__c]))[(int)(__c)])
                                                      ^
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:64:24: note: in expansion of macro ‘__ctype_lookup’
 #define isxdigit(__c) (__ctype_lookup(__c)&amp;(_X|_N))
                        ^~~~~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/json/json.c:133:10: note: in expansion of macro ‘isxdigit’
     if (!isxdigit(next(lexer))) {
          ^~~~~~~~
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:58:54: error: array subscript has type ‘char’ [-Werror=char-subscripts]
 #define __ctype_lookup(__c) ((__ctype_ptr__+sizeof(""[__c]))[(int)(__c)])
                                                      ^
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:64:24: note: in expansion of macro ‘__ctype_lookup’
 #define isxdigit(__c) (__ctype_lookup(__c)&amp;(_X|_N))
                        ^~~~~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/json/json.c:137:10: note: in expansion of macro ‘isxdigit’
     if (!isxdigit(next(lexer))) {
          ^~~~~~~~
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:58:54: error: array subscript has type ‘char’ [-Werror=char-subscripts]
 #define __ctype_lookup(__c) ((__ctype_ptr__+sizeof(""[__c]))[(int)(__c)])
                                                      ^
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:64:24: note: in expansion of macro ‘__ctype_lookup’
 #define isxdigit(__c) (__ctype_lookup(__c)&amp;(_X|_N))
                        ^~~~~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/json/json.c:141:10: note: in expansion of macro ‘isxdigit’
     if (!isxdigit(next(lexer))) {
          ^~~~~~~~
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:58:54: error: array subscript has type ‘char’ [-Werror=char-subscripts]
 #define __ctype_lookup(__c) ((__ctype_ptr__+sizeof(""[__c]))[(int)(__c)])
                                                      ^
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:64:24: note: in expansion of macro ‘__ctype_lookup’
 #define isxdigit(__c) (__ctype_lookup(__c)&amp;(_X|_N))
                        ^~~~~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/json/json.c:145:10: note: in expansion of macro ‘isxdigit’
     if (!isxdigit(next(lexer))) {
          ^~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/json/json.c: In function ‘lexer_number’:
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:58:54: error: array subscript has type ‘char’ [-Werror=char-subscripts]
 #define __ctype_lookup(__c) ((__ctype_ptr__+sizeof(""[__c]))[(int)(__c)])
                                                      ^
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:63:23: note: in expansion of macro ‘__ctype_lookup’
 #define isdigit(__c) (__ctype_lookup(__c)&amp;_N)
                       ^~~~~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/json/json.c:221:7: note: in expansion of macro ‘isdigit’
   if (isdigit(chr) || chr == '.') {
       ^~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/json/json.c: In function ‘lexer_json’:
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:58:54: error: array subscript has type ‘char’ [-Werror=char-subscripts]
 #define __ctype_lookup(__c) ((__ctype_ptr__+sizeof(""[__c]))[(int)(__c)])
                                                      ^
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:63:23: note: in expansion of macro ‘__ctype_lookup’
 #define isdigit(__c) (__ctype_lookup(__c)&amp;_N)
                       ^~~~~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/json/json.c:257:8: note: in expansion of macro ‘isdigit’
    if (isdigit(peek(lexer))) {
        ^~~~~~~
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:58:54: error: array subscript has type ‘char’ [-Werror=char-subscripts]
 #define __ctype_lookup(__c) ((__ctype_ptr__+sizeof(""[__c]))[(int)(__c)])
                                                      ^
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:65:23: note: in expansion of macro ‘__ctype_lookup’
 #define isspace(__c) (__ctype_lookup(__c)&amp;_S)
                       ^~~~~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/json/json.c:263:8: note: in expansion of macro ‘isspace’
    if (isspace(chr)) {
        ^~~~~~~
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:58:54: error: array subscript has type ‘char’ [-Werror=char-subscripts]
 #define __ctype_lookup(__c) ((__ctype_ptr__+sizeof(""[__c]))[(int)(__c)])
                                                      ^
/opt/sdk/zephyr-sdk-0.9.1/sysroots/armv5-zephyr-eabi/usr/include/ctype.h:63:23: note: in expansion of macro ‘__ctype_lookup’
 #define isdigit(__c) (__ctype_lookup(__c)&amp;_N)
                       ^~~~~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/json/json.c:268:8: note: in expansion of macro ‘isdigit’
    if (isdigit(chr)) {
        ^~~~~~~
cc1: all warnings being treated as errors
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/Makefile.build:183: recipe for target 'lib/json/json.o' failed
make[5]: *** [lib/json/json.o] Error 1
make[5]: Target '__build' not remade because of errors.
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/Makefile.build:357: recipe for target 'lib/json' failed
make[4]: *** [lib/json] Error 2
  CC      lib/libc/newlib/libc-hooks.o
  LD      lib/libc/newlib/built-in.o
  LD      lib/libc/built-in.o
make[4]: Target '__build' not remade because of errors.
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/Makefile:1039: recipe for target 'lib' failed
make[3]: *** [lib] Error 2
  CC      kernel/version.o
```

--- a/tests/lib/json/testcase.yaml
+++ b/tests/lib/json/testcase.yaml
@@ -1,5 +1,6 @@
 tests:
 -   test:
         build_only: true
+        filter: not CONFIG_NEWLIB_LIBC
         min_ram: 32
         tags: json

